tests/cloud/basicFlow.ts

```diff
@@ -0,0 +1,393 @@
+			userKeypairs.push(result.keypair);
+
+			const userATA = await getOrCreateATA(
+				cloudMint,
+				result.keypair.publicKey,
+				result.keypair,
+				provider.connection
+			);
+
+			await mintTo(
+				provider.connection,
+				keypair,
+				cloudMint,
+				userATA,
+				keypair.publicKey,
+				lockAmount.toNumber() * 2
+			);
+		}
+
+		await initializeLocker();
+	});
+
+	// Multiple user stake-unstake-cancel-unstake-withdraw scenarios
+	it("goes through the happy path", async () => {
+		// Initialize escrows for our users and let them stake the first time
+		for (const keypair of userKeypairs) {
+			const wallet = new Wallet(keypair);
+			const voterProgram = createLockedVoterProgram(wallet, LOCKED_VOTER_PROGRAM_ID);
+			const [escrow, _bump] = deriveEscrow(locker, wallet.publicKey, LOCKED_VOTER_PROGRAM_ID);
+
+			// Escrow is what each user starts to have on first stake
+			await voterProgram.methods
+				.newEscrow()
+				.accounts({
+					escrow,
+					escrowOwner: wallet.publicKey,
+					locker,
+					payer: wallet.publicKey,
+					systemProgram: web3.SystemProgram.programId,
+				})
+				.rpc();
+
+			// On the frontend probably just add create ata idempotent ix to the tx if user is new?
+			const escrowATA = await getOrCreateATA(cloudMint, escrow, keypair, provider.connection);
+
+			// Cloud ATA
+			const userATA = await getOrCreateATA(
+				cloudMint,
+				wallet.publicKey,
+				keypair,
+				provider.connection
+			);
+
+			let escrowAccountBefore = await voterProgram.account.escrow.fetch(escrow);
+
+			// Escrow amount in the beginning must be 0
+			assert(escrowAccountBefore.amount.toNumber() === 0, "Escrow initialized with amount > 0");
+			assert(escrowAccountBefore.isMaxLock === true, "Escrow isn't max locked by default");
+
+			await voterProgram.methods
+				.increaseLockedAmount(lockAmount)
+				.accounts({
+					escrow,
+					escrowTokens: escrowATA,
+					locker,
+					payer: voterProgram.provider.publicKey,
+					sourceTokens: userATA,
+					tokenProgram: TOKEN_PROGRAM_ID,
+				})
+				.rpc();
+
+			let escrowAccountAfter = await voterProgram.account.escrow.fetch(escrow);
+			const userATABalanceAfter = await provider.connection
+				.getTokenAccountBalance(userATA)
+				.then((b) => b.value.amount);
+			const escrowATABalanceAfter = await provider.connection
+				.getTokenAccountBalance(userATA)
+				.then((b) => b.value.amount);
+
+			// Set escrow data to staked amount
+			assert(
+				escrowAccountAfter.amount.toNumber() === lockAmount.toNumber(),
+				"Escrow amount incorrect after staking"
+			);
+			// Check if users have less tokens now
+			assert(userATABalanceAfter === String(lockAmount.toNumber()));
+			// Check if escrow ata has more tokens now
+			assert(escrowATABalanceAfter === String(lockAmount.toNumber()));
+		}
+
+		// Let users stake once more each (same process testing multiple staking attempts)
+		for (const keypair of userKeypairs) {
+			const wallet = new Wallet(keypair);
```

Comment:
nit: `staker` would be a better name imo

Reply:
and `keypair` to be `userKeypair` or `stakerKeypairs` and `stakerKeypair` respectively